lib/dashboard.js

```diff
@@ -7,118 +7,127 @@ var StreamView = require("./views/stream-view");
-  var lines = data.replace(/\n$/, "");
+  this.views.eventLoop = new EventLoopView({
+    parent: this.container,
+    layoutConfig: config.eventLoop
+  });
 
-  streamView.onEvent(lines);
-};
+  this.views.memory = new MemoryView({
+    parent: this.container,
+    layoutConfig: config.memory
+  });
 
-Dashboard.prototype._createStdoutView = function (container) {
-  var stdoutView = new StreamView({
-    parent: container,
-    scrollback: this.options.scrollback,
-    label: this.options.interleave ? "stdout / stderr" : "stdout",
-    color: this.options.interleave ? "light-blue" : "green",
-    getPosition: stdOutPosition
+  this.views.help = new HelpView({
+    parent: this.container
   });
 
-  this.eventPump.addListener("stdout", this._logStream.bind(this, stdoutView));
+  this.currentLayout = 0;
 };
 
-Dashboard.prototype._createStderrView = function (container) {
-  if (!this.options.interleave) {
-    var stderrView = new StreamView({
-      parent: container,
-      scrollback: this.options.scrollback,
-      label: "stderr",
-      color: "red",
-      getPosition: stdErrPosition
-    });
-
-    this.eventPump.addListener("stderr", this._logStream.bind(this, stderrView));
-  }
-};
+Dashboard.prototype._configureKeys = function () {
 
-Dashboard.prototype._createMetricsViews = function (container) {
-  _.each(metrics, function (Metric, index) {
-    var view = new Metric({
-      parent: container,
-      getPosition: metricsPosition,
-      index: index
-    });
-    this.eventPump.addListener("metrics", view.onEvent.bind(view));
+  this.screen.key(["left", "right"], function (ch, key) {
+    var delta = key.name === "left" ? -1 : 1;
+    var target = (this.currentLayout + delta + layouts.length) % layouts.length;
+    this._showLayout(target);
   }.bind(this));
-};
 
-Dashboard.prototype._createView = function () {
-  // fixes weird scrolling issue
-  var container = blessed.box({});
+  this.screen.key("?", this.views.help.node.toggle.bind(this.views.help.node));
 
-  this.screen.append(container);
+  this.screen.key("escape", function () {
+    if (this.views.help.node.visible) {
+      this.views.help.node.hide();
+    } else {
+      this._showLayout(0);
+    }
+  }.bind(this));
 
-  this._createStdoutView(container);
-  this._createStderrView(container);
+  this.screen.key(["q", "C-c"], function () {
+    process.exit(0); // eslint-disable-line no-process-exit
+  });
+};
 
-  this._createMetricsViews(this.screen);
+Dashboard.prototype.onEvent = function (event) {
+  this.screen.emit(event.type, event.data);
+  // avoid double screen render for stream events (Element calls screen.render on scroll)
+  // TODO dashboard shouldn't know which events are used by which widgets
+  if (event.type === "metrics") {
+    this.screen.render();
+  }
+};
 
+Dashboard.prototype._showLayout = function (id) {
+  if (this.currentLayout === id) {
+    return;
+  }
+  _.each(this.views, function (v) {
```

Comment:
`this.views` is all of the views, active or not. (`hide()` is just a noop on an already hidden view)

agree about layouts managing views. would be more clear if dashboard managed layouts which manage views - rather than current setup where dashboard manages layouts and views (layouts being essentially just config)